registry/src/extensions.rs

```diff
@@ -15,21 +17,74 @@ pub fn check_input(input: &str) -> Result<(), ExtensionRegistryError> {
 
 pub async fn add_extension_owner(
     extension_id: i64,
-    user_id: String,
+    user_id: &String,
+    user_name: &String,
     conn: Data<Pool<Postgres>>,
 ) -> Result<(), ExtensionRegistryError> {
     let mut tx = conn.begin().await?;
     sqlx::query!(
         "
-        INSERT INTO extension_owners(extension_id, owner_id, created_at, created_by)
-        VALUES ($1, $2, (now() at time zone 'utc'), $3)
+        INSERT INTO extension_owners(extension_id, owner_id, user_name, created_at, created_by)
+        VALUES ($1, $2, $3, (now() at time zone 'utc'), $2)
         ",
         extension_id as i32,
         user_id,
-        user_id
+        user_name,
     )
     .execute(&mut tx)
     .await?;
     tx.commit().await?;
     Ok(())
 }
+
+pub async fn extension_owners(
+    extension_name: &str,
+    conn: Data<Pool<Postgres>>,
+) -> Result<Vec<Value>, ExtensionRegistryError> {
+    let mut extension_owners: Vec<Value> = Vec::new();
+    // Create a transaction on the database
+    let mut tx = conn.begin().await?;
+    let ext = sqlx::query!("SELECT * FROM extensions WHERE name = $1", extension_name)
+        .fetch_one(&mut tx)
+        .await?;
+    let id: i32 = ext.id as i32;
```

Comment:
You might be able to get away with only selecting the data you're using.

```suggestion
    let ext_id = sqlx::query_as!(i32, "SELECT id FROM extensions WHERE name = $1", extension_name)
        .fetch_one(&mut tx)
        .await?;
    let id: i32 = ext.id as i32;
```